src/routers/routerV2.ts

```diff
@@ -1,12 +1,23 @@
 import { Router } from 'express';
+import rateLimit from 'express-rate-limit';
 import { getWords, getWord } from '../controllers/words';
 import { getExample, getExamples } from '../controllers/examples';
 import { getNsibidiCharacter, getNsibidiCharacters } from '../controllers/nsibidi';
 import { getTranscription } from '../controllers/speechToText';
+import { getTranslation } from '..//controllers/translation';
 import validId from '../middleware/validId';
 import validateApiKey from '../middleware/validateApiKey';
 import analytics from '../middleware/analytics';
 import attachRedisClient from '../middleware/attachRedisClient';
+import { MiddleWare } from 'src/types';
+
+const ONE_DAY = 24 * 60 * 60 * 1000;
+const REQUESTS_PER_MS_TRANSLATION = 5;
+
+const translationRateLimiter: MiddleWare = rateLimit({
```

Comment:
i think we want rate limiting to prevent too many subsequent requests in period of time rather than limiting the total number of requests that can be made. the logic that's responsible for capping the total number of requests that the current user can make with the API Key lives in `authorizeDeveloperUsage`. 

i do think we want a rate limiter in general - we can talk about that more. 

also, can we have this rate limiter not be applied if the incoming request comes from our official platforms? we would want to check to see if the `X-API-Key` header is equal to `MAIN_KEY` which is in `config.ts`

Reply:
After our discussion, I made this ticket for us to push a fix for rate limiting in a separate PR https://github.com/nkowaokwu/igbo_api/issues/814